tests/unit/test_util.py

```diff
@@ -86,32 +80,50 @@ def test_pairwise_difference(self) -> None:
         output = pairwise(difference)(x_array, y_array)
         assert jnp.linalg.norm(output - expected_output) == pytest.approx(0.0, abs=1e-3)
 
-    def test_apply_negative_precision_threshold_invalid(self) -> None:
-        """
-        Test apply_negative_precision_threshold for an invalid threshold.
-
-        A negative precision threshold is given, which should be rejected by the
-        function.
-        """
-        with pytest.raises(ValueError):
-            apply_negative_precision_threshold(x=0.1, precision_threshold=-1e-8)
+    @pytest.mark.parametrize(
+        "length",
+        (0, 1, -1, 2, 10),
+        ids=[
+            "zero",
+            "below_tree_length",
+            "negative",
+            "tree_length",
+            "above_tree_length",
+        ],
+    )
+    def test_tree_leaves_repeat(self, length: int) -> None:
+        """Test tree_leaves_repeat for various length parameters."""
+        tree = (None, 1)
+        tree_leaves = jtu.tree_leaves(tree, is_leaf=lambda x: x is None)
+        repeated_tree_leaves = tree_leaves_repeat(tree, length)
+        expected_leaves = tree_leaves + [
```

Comment:
I think it would be better to hardcode `tree_leaves` as `[None, 1]` here and delete line 97. Then, it also tests that `is_leaf` is behaving as expected.